### PR TITLE
Applies micro-optimizations and removes obsolete settings

### PR DESCRIPTION
**Pull request by sorvell**
Applies micro-optimizations that were found to improve element creation benchmarks by 5-10%, and removes obsolete settings:

* Removed `legacyNoBatch` and `legacyNotifyOrder` settings.
* dom-if/repeat: `dom-change` and `renderedCount` no longer fire with `legacyOptimizations` set.
* `LegacyElementMixin`: `isAttached` now set before calling `connectedCallback` so it is batched with initial rendering.
* `PropertiesChanged`: property accessor code now inlined for efficiency rather than calling `_get/_setProperty`. The `__dataCounter` tracking flag has been moved here to avoid the need to override `_flushProperties` in `PropertyEffects`.
* `PropertyEffects`: inlined `runEffectsForProperty` into `runEffects` for efficiency. Removed wrapping around sending data events.
* `async`: In the microtask scheduler, now only provoke a DOM mutation if needed.

### CODE REVIEW

File: lib/utils/settings.js [hunk 1]
```diff
@@ -278,3 +278,22 @@ export let fastDomIf = window.Polymer && window.Polymer.fastDomIf || false;
...
+ * Setting to disable `dom-change` and `rendered-item-count` events from
+ * `dom-if` and `dom-repeat`. Users can opt back into `dom-change` events by
+ * setting the `notify-dom-change` attribute (`notifyDomChange: true` property)
+ * to `dom-if`/`don-repeat` instances.
+ */
+export let suppressTemplateNotifications =
+  window.Polymer && window.Polymer.suppressTemplateNotifications || false;
+
+/**
+ * Sets `fastDomIf` globally, to put `dom-if` in a performance-optimized mode.
```
Comment (sorvell) [inline lib/utils/settings.js]: Fix docs, not `fastDomIF`

File: lib/mixins/property-effects.js [hunk 2]
```diff
@@ -320,8 +320,7 @@ function dispatchNotifyEvent(inst, eventName, value, path) {
   if (path) {
     detail.path = path;
   }
-  // Note, since this does not bubble, it should not need to be wrapped.
-  inst.dispatchEvent(new CustomEvent(eventName, { detail }));
+  wrap(/** @type {!HTMLElement} */(inst)).dispatchEvent(new CustomEvent(eventName, { detail }));
```
Comment (sorvell) [inline lib/mixins/property-effects.js]: Add comment about why this needs wrap (flushing)

File: lib/utils/settings.js (hunk 1 above)
Comment (kevinpschaaf) [reply lib/utils/settings.js]: Done

File: lib/mixins/property-effects.js (hunk 2 above)
Comment (kevinpschaaf) [reply lib/mixins/property-effects.js]: Done

**Pull request merged**